### DISCUSSION TITLE
http- vs. modbus-Abfrage / Vor-/Nachteile?

### DISCUSSION THREAD

**Discussion by jaydee73** [wiggal/GEN24_Ladesteuerung#78] [General]
Hallo zusammen,
da jetzt ja seit einigen Versionen auch die Abfrage via http möglich ist, wollte ich mal fragen, ob beide Methoden "gleichwertig" sind? Oder kann die eine Methode mehr als die Andere? In der ini steht ja bei einigen Optionen "Modbus" und bei den meisten Optionen "Http und Modbus". Bedeutet also, die Optionen, wo nur Modbus steht, gehen bei http-Abfrage nicht. Sind die da nicht notwendig?
Ich frage halt derzeit bei mir noch per Modbus ab und überlege, ob es Sinn macht, auf http zu switchen.

Gruß,
JD

**Comment by wiggal** [comment]
Hallo @jaydee73,
ich versuche mal einige deiner Fragen aus meiner Sicht zu beleuchten:

>Bedeutet also, die Optionen, wo nur Modbus steht, gehen bei http-Abfrage nicht. 

Ja, die Optionen bei denen nur Modbus steht, gehen im http-Aufruf aktuell nicht. Es gibt aber auch Funktionen (als Erstes die Eigenverbrauchs-Optimierung ab Version 0.21.3), die nur http-Aufruf gehen.
>Sind die da nicht notwendig?

Manche Optionen funktionieren halt nur in der jeweiligen Methode.
>Oder kann die eine Methode mehr als die Andere?

Fallback funktioniert nur im Modbus (nach meinem Wissensstand)
Eigenverbrauchs-Optimierung z.B. geht nur in HTTP. 
In HTTP sind grundsätzlich alle Optionen denkbar, die man im GEN24 in der Oberfläche einstellen kann.

**Hier noch Vorteile der HTTP-Methode, die mich veranlasst haben, diese zu bauen:**

Der wichtigste Grund ist, dass ich eine Einspeisebegrenzung (70%) habe, und ich daher versuchen muss möglichst viel Strom über der Grenze in den Akku zu bringen. 
Die Umgehung der Einspeisebegrenzung im Modbus, ist nur eine Krücke, die Systembedingt nicht ganz toll funktioniert.
Da der GEN24 die Produktion auf die Einspeisebegrenzung begrenzt, weis man nie wie viel wirklich produziert werden könnte.
Deshalb schaltet die Steuerung an der Einspeisebegrenzung auf Verdacht um den WRSchreibGrenze_nachOben hoch, reicht die nicht beim nächsten Durchlauf nochmal. Da die jedes Mal zu wenig oder zu viel sein kann, ist die Methode ungenau.
EDIT 11.07.24 >>>>
Wie ich gerade festgestellt habe, wird auch die Energie über der AC-Kapazität des GEN24 automatisch in den Akku gespeichert. Deshalb entfällt auch die Begrenzung auf die AC-Kapazität des GEN24 (WR_Kapazitaet).
<<<EDIT

Wenn man die Ladegrenze in der GEN24-Oberfläche setzt, regel er den Überschuss automatisch selbst. Wenn z.B. die maximale Ladegrenze noch auf 0W steht, die Produktion aber über die Einspeisebegrenzung geht, speichert der GEN24 den Überschuss trotzdem automatisch in den Akku. Diese Methode ist ziemlich genau und somit bekommt man mehr Überschuss in den Akku, bzw. wird kaum Strom durch Abregelung des GEN24 verschenkt.
Zusätzlich kann ich mit der Eigenverbrauchs-Optimierung nachts kleine Mengen aus dem Akku entnehmen, um bei guten Prognosen für den nächsten Tag Platz im Akku zu haben.
Diese Vorteile habe ich in der Modbus-Option  nicht.

Ich hoffe es bringt ein bisschen Klarheit in die Materie

Schönen Gruß
Wiggal
